spyrit/learning/model_Had_DCAN.py

```diff
@@ -700,24 +700,25 @@ def forward_reconstruct_mmse(self, x, b, c, h, w):
     def forward_preprocess(self, x, b, c, h, w):
         #- Pre-processing (use batch norm to avoid division by N0 ?)
         x = x[:,:,self.even_index] - x[:,:,self.uneven_index];
-        x = 2*x-torch.reshape(self.Patt(torch.ones(b*c,1, h,w).to(x.device)),(b*c,1,self.M));
+        x = 2*x-torch.reshape(self.Patt(torch.ones(b*c,1,h,w).to(x.device)),(b,c,self.M));
         return x
     
     def forward_maptoimage(self, x, b, c, h, w):
         #--Projection to the image domain
         x = self.fc1(x);
-        x = x.view(b*c,1,h,w)
+        x = x.view(b, c, h, w)
```

Comment:
Is this view necessary? x already (b, c, h, w) ?

Reply:
Yes, because `self.fc1(x)` returns an array with dimensions [b,c,h*w].